### PR TITLE
Feature/max depth

### PR DESCRIPTION
**Pull request by pedrorponga**

### CODE REVIEW

File: skfda/ml/classification/maximum_depth.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+        Returns:
+            (np.array): y : array of shape [n_samples] or
+            [n_samples, n_outputs] with class labels for each data sample.
+
+        """
+        sklearn_check_is_fitted(self, 'distributions_')
+
+        depths = [distribution.predict(X) for distribution in self.distributions_]
+        return np.array([self.classes_[i] for i in np.argmax(depths, axis=0)])
```
Comment (vnmabus) [inline skfda/ml/classification/maximum_depth.py:102]: As the array is unidimensional, it is not necessary to specify the axis. Also, you can index an `ndarray` using an array.

```suggestion
        return self.classes_[np.argmax(depths)]
```

File: skfda/ml/classification/maximum_depth.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        check_classification_targets(y)
+
+        le = LabelEncoder()
+        y_ind = le.fit_transform(y)
+        self.classes_ = classes = le.classes_
+        n_classes = classes.size
+        if n_classes < 2:
+            raise ValueError(f'The number of classes has to be greater than'
+                             f' one; got {n_classes} class')
+
```
Comment (vnmabus) [inline skfda/ml/classification/maximum_depth.py:80]: The above preprocessing (with the check included) seems useful for other classification methods. Maybe we should create a private function in utils to prevent code repetition.

**Review by vnmabus — CHANGES_REQUESTED** [review]

File: skfda/ml/classification/maximum_depth.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        self.distributions_ = []
```
[the remark this answers is not in the document]
Comment (pedrorponga) [reply skfda/ml/classification/maximum_depth.py]:
```suggestion
        self.distributions_ = [None]*n_classes
```

**Review by pedrorponga — COMMENTED** [review]
Private function in fit method needs to be created

**Review by vnmabus — CHANGES_REQUESTED** [review]

File: skfda/ml/classification/maximum_depth.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    See also
+    --------
+    :class:`~skfda.ml.classification.DD-plot`
+
+    """
+
+    def __init__(self, depth_method=IntegratedDepth()):
+        """Initialize the classifier."""
+        self.depth_method = depth_method
```
Comment (pedrorponga) [inline skfda/ml/classification/maximum_depth.py:61]: Don't forget to add :class:`~skfda.ml.classification.DDClasifier

Comment (pedrorponga) [inline skfda/ml/classification/maximum_depth.py:61]: ll

File: skfda/ml/classification/maximum_depth.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+           1, 1])
+    
+    Finally, we calculate the mean accuracy for the test data
+
+    >>> clf.score(X_test, y_test)
+    0.7916666666666666
+
+    See also
+    --------
+
```
Comment (pedrorponga) [inline skfda/ml/classification/maximum_depth.py:56]: Don't forget to add :class:`~skfda.ml.classification.DDClasifier

File: skfda/_utils/_utils.py [hunk 6]
```diff
@@ -447,3 +447,20 @@ def _check_estimator(estimator):
     instance = estimator()
     check_get_params_invariance(name, instance)
     check_set_params(name, instance)
+
+
+def _classifier_fit_init(y):
```
Comment (vnmabus) [inline skfda/_utils/_utils.py:452]: Maybe the name could be more clear.

```suggestion
def _classifier_get_classes(y):
```

File: skfda/ml/classification/maximum_depth.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+from skfda.exploratory.depth import *
+from skfda._utils import _classifier_get_classes
+
+class MaximumDepth(BaseEstimator, ClassifierMixin):
+    """Maximum depth classifier for functional data.
+
+    Test samples are classified to the class where they are deeper.
+
+    Parameters
+    ----------
```
Comment (vnmabus) [inline skfda/ml/classification/maximum_depth.py:17]: We use Google notation for docstrings, not Numpy notation. Please, change it.

File: skfda/ml/classification/maximum_depth.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+import numpy as np
+
+from sklearn.base import ClassifierMixin, BaseEstimator, clone
+from sklearn.utils.validation import check_is_fitted as sklearn_check_is_fitted
+
+from skfda.exploratory.depth import *
+from skfda._utils import _classifier_get_classes
+
+class MaximumDepth(BaseEstimator, ClassifierMixin):
```
Comment (pedrorponga) [inline skfda/ml/classification/maximum_depth.py:11]: MaximumDepthClassifier

File: skfda/ml/classification/maximum_depth.py [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    """Maximum depth classifier for functional data.
+
+    Test samples are classified to the class where they are deeper.
+
+    Parameters:
+        depth_method (callable, (default
```
Comment (pedrorponga) [inline skfda/ml/classification/maximum_depth.py:17]: Cambiar default

File: skfda/ml/classification/maximum_depth.py [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+        Args:
+            X (:class:`FDataGrid`): FDataGrid with the test samples.
+
+        Returns:
+            y (np.array): array of shape [n_samples] with class labels
+                for each data sample.
+
+        """
+        sklearn_check_is_fitted(self)
+
```
Comment (pedrorponga) [inline skfda/ml/classification/maximum_depth.py:87]: Pep8, Pep257

File: skfda/ml/classification/maximum_depth_classifier.py [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        array([1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1,
+                1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1])
+
+        Finally, we calculate the mean accuracy for the test data
+
+        >>> clf.score(X_test, y_test)
+        0.875
+
+    See also:
+
```
Comment (pedrorponga) [inline skfda/ml/classification/maximum_depth_classifier.py:53]: Don't forget to add DDClassifier

**Review by vnmabus — APPROVED** [review]
LGTM.

**Review by vnmabus — APPROVED** [review]
LGTM.
